imagekit-sdk/src/test/java/io/imagekit/sdk/ImageKitTest.java

```diff
@@ -2,629 +2,1890 @@
+		assertEquals(moveFileRequestJson, utf8RequestBody);
+		assertEquals("application/json; charset=utf-8", request.getHeader("Content-Type"));
+		assertEquals("POST /v1/files/move HTTP/1.1", request.getRequestLine());
+		assertEquals(RestClient.API_BASE_URL.concat("v1/files/move"), request.getRequestUrl().toString());
+	}
+
+	@Test(expected = NotFoundException.class)
+	public void renameFile_404_Expected() throws InterruptedException, IOException, ConflictException,
+			PartialSuccessException, NotFoundException, BadRequestException, InternalServerException, UnknownException,
+			ForbiddenException, TooManyRequestsException, UnauthorizedException {
+
+		RenameFileRequest renameFileRequest = new RenameFileRequest();
+		renameFileRequest.setFilePath("/sample_image_th.jpg");
+		renameFileRequest.setNewFileName("new_car.jpg");
+		renameFileRequest.setPurgeCache(true);
+
+		MockWebServer server = new MockWebServer();
+		server.enqueue(new MockResponse().setResponseCode(404)
+				.setBody("{\n"
+						+ "    \"message\": \"No file found in media library with filePath /sample_image_th.jpg\",\n"
+						+ "    \"help\": \"For support kindly contact us at support@imagekit.io .\",\n"
+						+ "    \"reason\": \"FILE_MISSING\"\n" + "}"));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		SUT.renameFile(renameFileRequest);
+		server.takeRequest();
+	}
+
+	@Test(expected = ConflictException.class)
+	public void renameFile_409_Expected() throws InterruptedException, IOException, ConflictException,
+			PartialSuccessException, NotFoundException, BadRequestException, InternalServerException, UnknownException,
+			ForbiddenException, TooManyRequestsException, UnauthorizedException {
+
+		RenameFileRequest renameFileRequest = new RenameFileRequest();
+		renameFileRequest.setFilePath("/new1.jpg");
+		renameFileRequest.setNewFileName("new_car.jpg");
+		renameFileRequest.setPurgeCache(true);
+
+		MockWebServer server = new MockWebServer();
+		server.enqueue(new MockResponse().setResponseCode(409)
+				.setBody("{\n"
+						+ "    \"message\": \"File with name new_car.jpg already exists at the same location.\",\n"
+						+ "    \"help\": \"For support kindly contact us at support@imagekit.io .\",\n"
+						+ "    \"reason\": \"FILE_ALREADY_EXISTS\"\n" + "}"));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		SUT.renameFile(renameFileRequest);
+		server.takeRequest();
+	}
+
+	@Test
+	public void renameFile_successExpected() throws InterruptedException, IOException, ConflictException,
+			PartialSuccessException, NotFoundException, BadRequestException, InternalServerException, UnknownException,
+			ForbiddenException, TooManyRequestsException, UnauthorizedException {
+
+		RenameFileRequest renameFileRequest = new RenameFileRequest();
+		renameFileRequest.setFilePath("/car_false.jpeg");
+		renameFileRequest.setNewFileName("new_car.jpeg");
+		renameFileRequest.setPurgeCache(true);
```

Comment:
Include test case for `false` condition as well `renameFileRequest.setPurgeCache(false);`